16/umbraco-cms/customizing/extending-overview/extension-types/menu-item.md

```diff
@@ -0,0 +1,268 @@
+---
+description: >-
+    Create menu items that appear throughout the backoffice, in sidebars, button flyouts, and more.
+---
+
+# Menu Items
+
+Menu Item extensions are used in combination with [Menu](menu.md) extensions. Menu items can be placed in custom menus, sidebars, and even added into the built-in/default Umbraco menu extensions. Extension authors can use the default Menu Item component or create their own custom Menu Items and register them as extensions.
+
+<figure><img src="../../../.gitbook/assets/menu-item.png" alt="" width="250"><figcaption><p>Menu Item</p></figcaption></figure>
+
+Menu Item extensions can be defined using either JSON in `umbraco-package.json` or using TypeScript.
+
+## Creating Menu Items
+
+Menu Item extensions can be created using either JSON or TypeScript. Both approaches are shown below.
+
+### Manifest
+
+To add custom menu items, you can define a single MenuItem manifest and link an element to it. In this element, you can fetch the data and render as many menu items as you want based on that data.
+
+{% tabs %}
+{% tab title="JSON" %}
+{% code title="umbraco-package.json" %}
+```json
+{
+    "$schema": "../../umbraco-package-schema.json",
+    "name": "My Package",
+    "version": "0.1.0",
+    "extensions": [
+        {
+            "type": "menuItem",
+            "alias": "My.MenuItem",
+            "name": "My Menu Item",
+            "element": "./menu-items.ts",
+            "meta": {
+                "label": "My Menu Item",
+                "menus": ["My.Menu"]
+            }
+        }
+    ]
+}
+```
+{% hint style="info" %}
+The `element` parameter is optional. Omitting it will render a menu item styled using Umbraco defaults.
+{% endhint %}
+{% endcode %}
+{% endtab %}
+{% tab title="TypeScript" %}
+
+Extension authors define the menu manifest, then register it dynamically/during runtime using a [Backoffice Entry Point](../../extending-overview/extension-types/backoffice-entry-point.md) extension.
+
+The `element` attribute will point toward a custom Lit component, an example of which will be in the next section of this article.
```

Comment:
It is unclear which `element`-attribute this refers to, as there is no element in the example. I also believe we should write that it should point to a custom web component rather than a Lit component.